src/redfishpower/redfishpower.c

```diff
@@ -754,6 +757,112 @@ static void setpowerpath(char **av, char **path, char **postdata)
     }
 }
 
+static void remove_initial_plugs(void)
+{
+    hostlist_iterator_t itr;
+    char *hostname;
+
+    if (!initial_plugs_setup)
+        return;
+
+    if (!(itr = hostlist_iterator_create(hosts)))
+        err_exit(true, "hostlist_iterator_create");
+
+    while ((hostname = hostlist_next(itr))) {
+        plugs_remove(plugs, hostname);
+        free(hostname);
+    }
+
+    hostlist_iterator_destroy(itr);
+    initial_plugs_setup = 0;
+    return;
+}
+
+static void setplugs(char **av)
+{
+    hostlist_t lplugs = NULL;
+    hostlist_t indices = NULL;
+    int plugcount, indexcount;
+    hostlist_iterator_t lplugsitr = NULL;
+    hostlist_iterator_t indicesitr = NULL;
+    char *plug;
+    char *indexstr;
+
+    if (!av[0] || !av[1]) {
+        printf("Usage: setplugs <plugnames> <hostindices>\n");
+        return;
+    }
+
+    if (!(lplugs = hostlist_create(av[0]))) {
+        printf("illegal plugs input\n");
+        goto cleanup;
+    }
+    if (!(indices = hostlist_create(av[1]))) {
+        printf("illegal index input\n");
+        goto cleanup;
+    }
+
+    if ((plugcount = hostlist_count(lplugs)) < 0)
+        err_exit(true, "hostlist_count error");
+    if ((indexcount = hostlist_count(indices)) < 0)
+        err_exit(true, "hostlist_count error");
+    if (plugcount != indexcount) {
+        printf("plugs count not equal to index count");
+        goto cleanup;
+    }
+
+    if (!(lplugsitr = hostlist_iterator_create(lplugs)))
+        err_exit(true, "hostlist_iterator_create");
+    if (!(indicesitr = hostlist_iterator_create(indices)))
+        err_exit(true, "hostlist_iterator_create");
+
+    /* if user is electing to configure their own plugs, we must remove
+     * all of the initial ones configured in setup_hosts()
+     */
+    remove_initial_plugs();
+
+    while ((plug = hostlist_next(lplugsitr))
+           && (indexstr = hostlist_next(indicesitr))) {
+        char *host;
+        char *endptr;
+        int index;
+
+        errno = 0;
+        index = strtol (indexstr, &endptr, 10);
+        if (errno
+            || endptr[0] != '\0'
+            || index < 0) {
+            printf("invalid index %s specified\n", indexstr);
+            free(plug);
+            free(indexstr);
+            goto cleanup;
+        }
```

Comment:
It might clean this block up a bit if
- instead of two iterators, just do `for (int i = 0; i < plugcount; i++)`
- use hostlist_nth() instead of the iterator values
- add a function for looking up a host from the nth host index

Reply:
ugh I'm dumb ... later on in the PR series I create a "setup_plug()" function.  Let me just bring that down now and it'll clean up the code a lot.